src/ctap/data_formats.rs

```diff
@@ -477,7 +512,7 @@ impl TryFrom<cbor::Value> for PublicKeyCredentialSource {
         use cbor::{SimpleValue, Value};
 
         let fields = read_array(&cbor_value)?;
-        if fields.len() != 6 {
+        if fields.len() != 7 {
```

Comment:
What happens if you advertise credProtect but the other side decides to not use the option? Isn't the array only going to be with 6 elements?

Reply:
The array always has 7 elements (for a given version of OpenSK). If credProtect is not used, the last element is the CBOR `null` value. The issue would be using a more recent OpenSK with an old storage. But a fix has already been merged. So after syncing to master this test disappears since we would use a map instead of an array.

Reply:
Actually, it is currently not possible to use a recent OpenSK with an old storage since we still use writable flash regions. So this test (length of array) cannot fail for 2 reasons:
- the fix to support old storage with new code is submitted
- the storage is still inside the app so a new app cannot use an old storage

Reply:
Please check the code again after the merge with master.

Reply:
Did you forget to push or is the merge with master still in progress? I don't see any new commits.